swarm/api/manifest_test.go

```diff
@@ -60,32 +61,55 @@ func checkEntry(t *testing.T, path, match string, trie *manifestTrie) {
 		t.Errorf("incorrect entry retrieved for '%s'. expected path '%v', got '%s'", path, match, fullpath)
 	}
+
+	if multiple && entry.Status != http.StatusMultipleChoices {
+		t.Errorf("Expected %d Multiple Choices Status for path %s, match %s, got %d", http.StatusMultipleChoices, path, match, entry.Status)
+	} else if !multiple && entry != nil && entry.Status == http.StatusMultipleChoices {
+		t.Errorf("Were not expecting %d Multiple Choices Status for path %s, match %s, but got it", http.StatusMultipleChoices, path, match)
+	}
 }
 
 func TestGetEntry(t *testing.T) {
 	// file system manifest always contains regularized paths
-	testGetEntry(t, "a", "a", "a")
-	testGetEntry(t, "b", "-", "a")
-	testGetEntry(t, "/a//", "a", "a")
+	testGetEntry(t, "a", "a", false, "a")
+	testGetEntry(t, "b", "-", false, "a")
+	testGetEntry(t, "/a//", "a", false, "a")
 	// fallback
-	testGetEntry(t, "/a", "", "")
-	testGetEntry(t, "/a/b", "a/b", "a/b")
+	testGetEntry(t, "/a", "", false, "")
+	testGetEntry(t, "/a/b", "a/b", false, "a/b")
 	// longest/deepest math
-	testGetEntry(t, "read", "read", "readme.md", "readit.md")
-	testGetEntry(t, "rf", "-", "readme.md", "readit.md")
-	testGetEntry(t, "readme", "readme", "readme.md")
-	testGetEntry(t, "readme", "-", "readit.md")
-	testGetEntry(t, "readme.md", "readme.md", "readme.md")
-	testGetEntry(t, "readme.md", "-", "readit.md")
-	testGetEntry(t, "readmeAmd", "-", "readit.md")
-	testGetEntry(t, "readme.mdffff", "-", "readme.md")
-	testGetEntry(t, "ab", "ab", "ab/cefg", "ab/cedh", "ab/kkkkkk")
-	testGetEntry(t, "ab/ce", "ab/ce", "ab/cefg", "ab/cedh", "ab/ceuuuuuuuuuu")
-	testGetEntry(t, "abc", "abc", "abcd", "abczzzzef", "abc/def", "abc/e/g")
-	testGetEntry(t, "a/b", "a/b", "a", "a/bc", "a/ba", "a/b/c")
-	testGetEntry(t, "a/b", "a/b", "a", "a/b", "a/bb", "a/b/c")
-	testGetEntry(t, "//a//b//", "a/b", "a", "a/b", "a/bb", "a/b/c")
+	testGetEntry(t, "read", "read", true, "readme.md", "readit.md")
+	testGetEntry(t, "rf", "-", false, "readme.md", "readit.md")
+	testGetEntry(t, "readme", "readme", false, "readme.md")
+	testGetEntry(t, "readme", "-", false, "readit.md")
+	testGetEntry(t, "readme.md", "readme.md", false, "readme.md")
+	testGetEntry(t, "readme.md", "-", false, "readit.md")
+	testGetEntry(t, "readmeAmd", "-", false, "readit.md")
+	testGetEntry(t, "readme.mdffff", "-", false, "readme.md")
+	testGetEntry(t, "ab", "ab", true, "ab/cefg", "ab/cedh", "ab/kkkkkk")
+	testGetEntry(t, "ab/ce", "ab/ce", true, "ab/cefg", "ab/cedh", "ab/ceuuuuuuuuuu")
+	testGetEntry(t, "abc", "abc", true, "abcd", "abczzzzef", "abc/def", "abc/e/g")
+	testGetEntry(t, "a/b", "a/b", true, "a", "a/bc", "a/ba", "a/b/c")
+	testGetEntry(t, "a/b", "a/b", false, "a", "a/b", "a/bb", "a/b/c")
+	testGetEntry(t, "//a//b//", "a/b", false, "a", "a/b", "a/bb", "a/b/c")
 }
+
+func TestExactMatch(t *testing.T) {
+	quitC := make(chan bool)
+	mf := manifest("shouldBeExactMatch.css", "shouldBeExactMatch.css.map")
+	trie, err := readManifest(mf, nil, nil, quitC)
```

Comment:
why the .map one?

Reply:
Because this is exactly what caused the error we are trying to fix: if there was a file "shouldBeExactMatch.css" and a file "shouldBeExactMatch.css.map" (.map files are often generated by SASS generators for CSS and are actually used in development only, but that's how we found out: our own webpage would not load...), then if the client requested "shouldBeExactMatch.css", the result would be 300 Multiple Choices, instead of serving the file itself

Reply:
Ok i get you i will be trying